geode-cq/src/distributedTest/java/org/apache/geode/cache/query/cq/CQMetricsDUnitTest.java

```diff
@@ -0,0 +1,169 @@
+        .statusIsSuccess();
+    qs.newCq("Select * from /region r where r.ID = 1", cqa).execute();
+
+    server1.invoke(() -> populateRegion(0, 100));
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(2));
+    });
+
+    // stop cq
+    qs.stopCqs();
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(0));
+    });
+
+    checkActiveCqCount(server1, 0);
+    checkActiveCqCount(server2, 0);
+  }
+
+  @Test
+  public void testCloseCq() throws Exception {
+    gfsh.executeAndAssertThat("create region --name=region --type=PARTITION")
+        .statusIsSuccess();
+    qs.newCq("Select * from /region r where r.ID = 1", cqa).execute();
+
+    server1.invoke(() -> populateRegion(0, 100));
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(2));
+    });
+
+    // close cq
+    qs.closeCqs();
+
+    locator.invoke(() -> {
+      Cache cache = getCache();
+      ManagementService service = ManagementService.getManagementService(cache);
+      DistributedSystemMXBean dsmbean = service.getDistributedSystemMXBean();
+      await().untilAsserted(() -> assertThat(dsmbean.getActiveCQCount()).isEqualTo(0));
+    });
+  }
+
+  private class TestCqListener implements CqListener, Serializable {
+    public int onEventCalls = 0;
+
+    @Override
+    public void onEvent(CqEvent aCqEvent) {
+      onEventCalls++;
+    }
+
+    @Override
+    public void onError(CqEvent aCqEvent) {}
+
+    @Override
+    public void close() {}
+  }
+
+  private static void populateRegion(int startingId, int endingId) {
+    Region exampleRegion = getCache().getRegion("region");
+    for (int i = startingId; i < endingId; i++) {
+      exampleRegion.put("" + i, new Portfolio(i));
+    }
+  }
+
+  private ClientCache createClientCache(Integer locator1Port) {
+    ClientCacheFactory ccf = new ClientCacheFactory();
+    ccf.addPoolLocator("localhost", locator1Port);
+    ccf.setPoolSubscriptionEnabled(true);
+    return ccf.create();
+  }
+
+  private void checkActiveCqCount(MemberVM vm, int expectedResult) {
+    vm.invoke(() -> {
+      QueryService queryService = getCache().getQueryService();
+      CqServiceStatistics cqServiceStats = queryService.getCqStatistics();
+      await()
```

Comment:
Please provide a timeout. I tested it with `timeout(30, TimeUnit.SECONDS)`. This way, this test will terminate in a "reasonable" time.